tests/src/Unit/AvPortalClientTest.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\media_avportal\src\Unit;
+
+use Drupal\media_avportal\AvPortalClient;
+use Drupal\Tests\UnitTestCase;
+
+/**
+ * Tests the AvPortalClient class.
+ */
+class AvPortalClientTest extends UnitTestCase {
```

Comment:
Since now this test needs to be reworked because we cannot test the output of the protected function, let's do this.
Create mock services as you did and put them in local vars.
Create a http client mock also inside the valid cases test method, and mock the proper method to expect the test data passed (see [here](https://www.drupal.org/docs/8/phpunit/comparison-with-phpunit-mocks) for examples).
Then have a separate method as you did for the exceptions cases, where you will instantiate a class with the mocked client of the setup.